### PR TITLE
Alerting: Query time logging

### LINKED ISSUE #56736: FE analytics: log query time in Alerting
We'd like to monitor what API queries take a long time (and tie up the UI), so we can debug issues without waiting for customers to run into a painful UX.

Can we easily hook up the JS Agent into RTK Query and detect when queries have completed (and returned data)?

Tracked in https://github.com/grafana/gops-frontend-squad/issues/1

### PR DESCRIPTION
**Pull request by konrad147**
**What does this PR**
This PR introduces basic logging for requests query time in Alerting
Eventually, we'd like to use tracing for query time tracking. Unfortunately, Grafana JS Agent doesn't support it currently due to OpenTelemetry issues.

**Which issue(s) does this PR fix?**:
Fixes #56736 

**Special notes for your reviewer**:
Logs created by this PR can be easily searched by LogQL

Sample LogQL query that selects requests taking more than a second to complete
```logql
{app="grafana-local", kind="log"} 
  | logfmt 
  | context_module = "Alerting" | context_loadTimeMs > 1000
  | line_format "{{.message}} | Time {{.context_loadTimeMs}}ms"
```
Query result
image

### CODE REVIEW

File: public/app/features/alerting/unified/state/actions.ts [hunk 1]
```diff
@@ -292,8 +327,11 @@ export const fetchRulesSourceBuildInfoAction = createAsyncThunk(
 
 export function fetchAllPromAndRulerRulesAction(force = false): ThunkResult<Promise<void>> {
   return async (dispatch, getStore) => {
+    const allStartLoadingTs = performance.now();
```
Comment (VikaCep) [inline public/app/features/alerting/unified/state/actions.ts:330]: In other cases you abstracted the performance code within the `logInfo` function, was it not possible to do something similar in this case?

File: public/app/features/alerting/unified/Analytics.ts [hunk 2]
```diff
@@ -8,3 +11,30 @@ export const LogMessages = {
...
+export function logInfo(message: string, context: Record<string, string | number> = {}) {
+  if (config.grafanaJavascriptAgent.enabled) {
+    agent.api.pushLog([message], {
+      level: GrafanaLogLevel.INFO,
+      context: { ...context, module: 'Alerting' },
+    });
+  }
+}
+
+export function withPerfLogging<TFunc extends (...args: any[]) => Promise<any>>(
```
Comment (VikaCep) [inline public/app/features/alerting/unified/Analytics.ts:25]: maybe use `unknown` instead of `any`?

Comment (konrad147) [reply public/app/features/alerting/unified/Analytics.ts:25]: Unfortunately, it doesn't work with `unknown`. `any` is not a problem in this case, all types are properly infered

File: public/app/features/alerting/unified/state/actions.ts (hunk 1 above)
Comment (konrad147) [reply public/app/features/alerting/unified/state/actions.ts:330]: It's technically possible but I think readability would suffer significantly. I'd have to wrap `Promise.allSettled` into a function wrapped into `withPerfLoggin`. I think it's not worth it

File: public/app/features/alerting/unified/Analytics.ts (hunk 2 above)
Comment (konrad147) [inline public/app/features/alerting/unified/Analytics.ts:25]: @ashharrison90 I believe there is no better way to handle `any` types there (If there is, I'm happy to implement).  
Is there a way to bypass betterer check for this line?

**Review by soniaAguilarPeiron — APPROVED** [review]
Nice!!🎉
I added a non-blocking comment

File: public/app/features/alerting/unified/Analytics.ts (hunk 2 above)
Comment (soniaAguilarPeiron) [inline public/app/features/alerting/unified/Analytics.ts:25]: Could we use another more explanatory name for this method?, not sure if something like `withPerformaceLogging`?

**Review by VikaCep — APPROVED** [review]
Great work! 💯

File: public/app/features/alerting/unified/Analytics.ts (hunk 2 above)
Comment (konrad147) [reply public/app/features/alerting/unified/Analytics.ts:25]: Exactly!

Comment (ashharrison90) [reply public/app/features/alerting/unified/Analytics.ts:25]: hey @konrad147, sorry for the late reply was at an offsite 😅 

given that the actual types when using this seem fine, i don't think there's a problem with the `any` here. the betterer rule is there to encourage us to think about these things, and 95% of the time we shouldn't need `any`, but think in this case it makes sense 😄 

you can ignore it via something like `// eslint-disable-next-line @typescript-eslint/no-explicit-any`

Comment (konrad147) [reply public/app/features/alerting/unified/Analytics.ts]: Thanks!

**Pull request merged**